### ISSUE TITLE
Space in directory structure fails plugin binary

### ISSUE THREAD

**Issue by electrical** [elastic/elasticsearch#12504]
When one of the parent directories has a space in there the plugin binary can't find the correct path.

```
$ pwd
/home/richard/temp dir/elasticsearch-1.7.0
$ ./bin/plugin
Error: Could not find or load main class dir.elasticsearch-1.7.0.config
```

**Comment by nik9000** [comment]
Confirmed on OSX and Ubuntu.

**Comment by nik9000** [comment]
Assigned to myself as I sent a PR for it. Anyone else can take it if they want it and think they can make a better pull request.

**Comment by nik9000** [comment]
Ok - we have a fix merged for 1.7 and now I'll need to forward port that to master.
